### PR TITLE
Improve atmos components code + OmegaStation fixes

### PR DESCRIPTION
**Pull request by pubby**
ree

Atmos components were all handling `on`+`stat` vars differently, and this was causing filters+mixers to remain shut off when APC power cycled them on/off. I have my suspicions that #30305 was caused by one of these filters turning off without the player realizing it. 

All stat checks have been changed to <s>`stat & (NOPOWER|BROKEN)`</s> `is_operational`, which from what I understand is more correct than ` stat & NOPOWER`. Filters+mixers now turn back on when power returns.

Unwrench checks have all been made consistent. Previously you could unwrench filters while on, but not pumps. Now you must turn filters+mixers off before unwrenching.

I also did some code cleanup removing a bunch of unchecked+inconsistent return values. This should not effect anything.

And then I removed stacked pipes on OmegaStation which were causing a bunch of runtimes.

### CODE REVIEW

**Review by Cyberboss — CHANGES_REQUESTED** [review]
All of the stat checks should be calls to is_operational() instead

File: code/modules/atmospherics/machinery/components/unary_devices/oxygen_generator.dm [hunk 1]
```diff
@@ -19,7 +19,7 @@
 	if(showpipe)
 		add_overlay(getpipeimage('icons/obj/atmospherics/components/unary_devices.dmi', "scrub_cap", initialize_directions)) //it works for now
 
-	if(!NODE1 || !on || stat & BROKEN)
+	if(!NODE1 || !on || (stat & BROKEN))
```
Comment (Cyberboss) [inline code/modules/atmospherics/machinery/components/unary_devices/oxygen_generator.dm]: is_operational()

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by pubby** [comment]
:ok_man:

File: code/modules/atmospherics/machinery/components/binary_devices/volume_pump.dm [hunk 2]
```diff
@@ -34,18 +34,16 @@ Thus, the two variables affect pump operation are set in New():
...
+	if(!is_operational())
 		icon_state = "volpump_off"
 		return
 
 	icon_state = "volpump_[on?"on":"off"]"
 
 /obj/machinery/atmospherics/components/binary/volume_pump/process_atmos()
 //	..()
-	if(stat & (NOPOWER|BROKEN))
+	if(!is_operational() || !on)
```
Comment (AnturK) [inline code/modules/atmospherics/machinery/components/binary_devices/volume_pump.dm]: cheaper/more common check first please.

File: code/modules/atmospherics/machinery/components/trinary_devices/filter.dm [hunk 3]
```diff
@@ -43,27 +43,19 @@
 	..()
 
 /obj/machinery/atmospherics/components/trinary/filter/update_icon_nopipes()
-
-	if(!(stat & NOPOWER) && on && NODE1 && NODE2 && NODE3)
+	if(is_operational() && on && NODE1 && NODE2 && NODE3)
```
Comment (AnturK) [inline code/modules/atmospherics/machinery/components/trinary_devices/filter.dm]: Same here.

File: code/modules/atmospherics/machinery/components/trinary_devices/filter.dm [hunk 4]
```diff
@@ -43,27 +43,19 @@
...
 	icon_state = "filter_off[flipped?"_f":""]"
 
 /obj/machinery/atmospherics/components/trinary/filter/power_change()
-	var/old_stat = stat
 	..()
-	if(stat & NOPOWER)
-		on = FALSE
-	if(old_stat != stat)
-		update_icon()
+	update_icon()
```
Comment (AnturK) [inline code/modules/atmospherics/machinery/components/trinary_devices/filter.dm]: Why did you remove the unchanged check ? power change is not always affecting every machine. Eg. light channel change won't affect atmos machinery but will still update them.

Comment (pubby) [reply code/modules/atmospherics/machinery/components/trinary_devices/filter.dm]: The other components don't have these checks so I figured I should make filters act the same.

You think having all of them check for unchanged would be better?

Comment (AnturK) [reply code/modules/atmospherics/machinery/components/trinary_devices/filter.dm]: In this case it was there because update_icon did some icon operations which can get expensive, but probably wouldn't hurt in general.
